packages/multichain/src/caip25Permission.test.ts

```diff
@@ -644,9 +697,100 @@ describe('caip25CaveatBuilder', () => {
     );
   });
 
-  it('throws if the eth accounts specified in the internal scopeObjects are not found in the wallet keyring', () => {
+  it('asserts the required accounts are supported', () => {
+    MockScopeSupported.isSupportedScopeString.mockReturnValue(true);
+    MockScopeSupported.isSupportedAccount.mockReturnValue(true);
+
+    try {
+      validator({
+        type: Caip25CaveatType,
+        value: {
+          requiredScopes: {
+            'eip155:1': {
+              accounts: ['eip155:1:0xdead'],
+            },
+            'bip122:000000000019d6689c085ae165831e93': {
+              accounts: ['bip122:000000000019d6689c085ae165831e93:123'],
+            },
+          },
+          optionalScopes: {
+            'eip155:5': {
+              accounts: ['eip155:5:0xbeef'],
+            },
+            'bip122:12a765e31ffd4059bada1e25190f6e98': {
+              accounts: ['bip122:12a765e31ffd4059bada1e25190f6e98:456'],
+            },
+          },
+          isMultichainOrigin: true,
+        },
+      });
+    } catch (err) {
+      // noop
+    }
+    expect(MockScopeSupported.isSupportedAccount).toHaveBeenCalledWith(
+      'eip155:1:0xdead',
+      {
+        getEvmInternalAccounts: expect.any(Function),
+        getNonEvmAccountAddresses: expect.any(Function),
+      },
```

Comment:
Yeah this pattern is a bit funny, passing these hooks to the utility helpers.... ah well

Reply:
yeah... not my favorite